include/sound/sof/virtio.h

```diff
@@ -126,4 +128,74 @@ struct dsp_sof_data_resp {
 #define HDR_SIZE_REQ offsetof(struct dsp_sof_data_req, data)
 #define HDR_SIZE_RESP offsetof(struct dsp_sof_data_resp, data)
 
+struct snd_sof_dev;
+struct sof_ipc_stream_posn;
+
+#if IS_ENABLED(CONFIG_VHOST_SOF)
+struct firmware;
+
+struct vhost_dsp;
+struct sof_vhost_ops {
+	int (*update_posn)(struct vhost_dsp *dsp,
+			   struct sof_ipc_stream_posn *posn);
+};
```

Comment:
do you foresee that we might have other ops?

 In the case of the mmap buffers then this will be a NULL ops, or an ops with no callback?